Wootric/segment-android-integration-wootric
# Update libraries to latest version

This will use will use the latest version of the Wootrick Android SDK which contains, amongst other things, new features like opt-out and is more secure.

## Changes
- Update Analytics to version 4.3.1
- Update WootricSDK to version 2.8.0

## Trello
https://trello.com/c/BVwjtqZS/19-05-gdpr-segment-updates

## How to test

1 - `git clone https://github.com/Wootric/segment-android-integration-wootric.git`
2 - `git checkout ds-update-integration`
3 - Create a new project in Android Studio with an empty activity
4 - Add a new MainApplication class
5 - Now modify the manifest to add Internet permission and assign our newly created class as the main application class.
The manifest should look something like this:
```xml
...
    <uses-permission android:name="android.permission.INTERNET"/>

    <application
        android:name=".MainApplication"
        ...
```

6 - Create a new Segment account if you don't have one already.
7 - Create a new Workspace
8 - Add the Wootric destination and set it with the account token and client id (client secret is not needed anymore)
9 - Once everything is setup in Segment we can now modify your MainApplication class to look something like this

```java
import com.segment.analytics.Analytics;
import com.wootric.analytics.android.integrations.wootric.WootricIntegration;

public class MainApplication extends Application {

    Analytics analytics;

    @Override
    public void onCreate() {
        super.onCreate();

        analytics = new Analytics.Builder(this, "write_key")
                .use(WootricIntegration.FACTORY)
                .build();
    }

    public Analytics getAnalytics() {
        return analytics;
    }
}
```

Your `write_key` can be found in Segment https://app.segment.com/organization/sources/source/settings/keys were you need to replace `organization` and `source` with your own values.

10 - Now we need to modify gradle to add our dependencies. Since we're working with a local dependency we need to modify `settings.gradle`, it should look something like this
```
include ':app'

include 'analytics-integration-wootric'
project(':analytics-integration-wootric').projectDir = new File('/Users/username/dir/segment-android-integration-wootric')
```
where `/Users/username/dir/segment-android-integration-wootric` should be the result of `pwd` from where you cloned the repo

11 -  Now let's modify the `build.gradle` (the one that belongs to the app e.g. `Module: app`) by adding these

```
dependencies {
    
    ...

    implementation 'com.segment.analytics.android:analytics:4.3.1'
    implementation 'com.wootric:wootric-sdk-android:2.8.0'

    implementation project(':analytics-integration-wootric')
}
```

12 - To finish, we need to add the logic to show the survey in `MainActivity`
```java

import com.segment.analytics.Analytics;
import com.wootric.analytics.android.integrations.wootric.WootricIntegration;
import com.wootric.androidsdk.Wootric;

public class MainActivity extends FragmentActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onResume() {
        super.onResume();

        Analytics analytics = ((MainApplication) getApplication()).getAnalytics();
        analytics.onIntegrationReady(WootricIntegration.FACTORY.key(), new Analytics.Callback<Wootric>() {
            @Override
            public void onReady(Wootric wootric) {
                // Set all aptional configuration here like:
                 wootric.setSurveyImmediately(true);
                // wootric.setLanguageCode("PL");
                wootric.survey();
            }
        });
    }
}
```
